skimage/_shared/_geometry.py

```diff
@@ -53,3 +61,81 @@ def polygon_area(pr, pc):
     pr = np.asarray(pr)
     pc = np.asarray(pc)
     return 0.5 * np.abs(np.sum((pc[:-1] * pr[1:]) - (pc[1:] * pr[:-1])))
+
+
+def _clip_sutherland_hodgman(rp, cp, rclip, cclip):
```

Comment:
Where does this implementation come from? These clipping algorithms are notoriously tricky to get right.

Reply:
I wrote it adapting the code you referred to I believe.

https://github.com/stefanv/supreme/blob/master/supreme/ext/polygon.c#L121